tests/test_base_scanner.py

```diff
@@ -112,13 +114,6 @@ def test_empty_issue_list_causes_scan(self, mock_scan: mock.MagicMock):
         list(test_scanner.issues)  # pylint: disable=pointless-statement
         mock_scan.assert_called()
 
-    @mock.patch("tartufo.scanner.ScannerBase.scan")
-    def test_scanner_does_not_rescan(self, mock_scan: mock.MagicMock):
-        test_scanner = TestScanner(self.options)
-        test_scanner._completed = True  # pylint: disable=protected-access
-        test_scanner.issues  # pylint: disable=pointless-statement
-        mock_scan.assert_not_called()
-
```

Comment:
Do we no longer have this guarantee? If we do, can we get this test added back? If not... why not?

Reply:
This test is literally testing whether a boolean is evaluated correctly, and it does so by making an assumption that when you call issues, the scan() function is not called.  Yet the scan() function itself guarantees that it won't re-scan when already completed.  This is a poor test that makes assumption about the behavior of the implementation to validate that a protected boolean field is honored.

Reply:
Yeah I agree that the state the test has evolved to is not great. Sounds like we'll need to add another test (external to this PR) to test that functionality in `scan()`, if one doesn't exist already.

Reply:
I would have phrased this slightly differently... "Yet the scan() function itself guarantees that it won't re-scan when already completed." True. The point of the test is to verify that guarantee is being honored.

Reply:
I think the # pylint: disable=protected-access should be a code smell that this test is excessively coupled with the implementation.  This is effectively sub-unit testing.